exporter/elasticsearchexporter/exporter_test.go

```diff
@@ -82,6 +82,39 @@ func TestExporterLogs(t *testing.T) {
 		rec.WaitItems(1)
 	})
 
+	t.Run("publish with bodymap encoding", func(t *testing.T) {
+		rec := newBulkRecorder()
+		server := newESTestServer(t, func(docs []itemRequest) ([]itemResponse, error) {
+			assert.JSONEq(t,
+				`{"@timestamp":"2024-03-12T20:00:41.123456789Z","id":1,"key":"value"}`,
+				string(docs[0].Document),
+			)
+			rec.Record(docs)
+			return itemsAllOK(docs)
+		})
+
+		exporter := newTestLogsExporter(t, server.URL, func(cfg *Config) {
+			cfg.Mapping.Mode = "bodymap"
+		})
+
+		logs := plog.NewLogs()
+		resourceLogs := logs.ResourceLogs().AppendEmpty()
+		scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+		logRecords := scopeLogs.LogRecords()
+		observedTimestamp := pcommon.Timestamp(time.Now().UnixNano())
+		logRecord := logRecords.AppendEmpty()
+
+		logRecord.SetObservedTimestamp(observedTimestamp)
+		bodyMap := pcommon.NewMap()
+		bodyMap.PutStr("@timestamp", "2024-03-12T20:00:41.123456789Z")
+		bodyMap.PutInt("id", 1)
+		bodyMap.PutStr("key", "value")
```

Comment:
can you try a nested object, or even better, with dot in the field name, and a case with `a:1` and `a:{b:1}`?

Reply:
I have added more test cases as requested, can you confirm if they are enough? Thanks.

Reply:
Thinking more about it, it is wise to make sure we have test cases for all Beats usages as well https://github.com/elastic/beats/pull/40974/files#diff-8b26a9726038f0fb32645af4e5a72c2a04c9f0c82d0757b004731e72f648e247R116-R255.

Reply:
I added test cases for an inner map, nested map and slice as well to make sure we cover all possible types that can be inside the log record body. While working on the test cases, I found that objmodel is still adding dots  to the key for the nested map case, which made me rethink my approach. Please see discussion in the other [thread](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35637#discussion_r1792169361) for context. Let me know if you have any additional concers regarding the test cases. Thank you!